### ISSUE TITLE
Funds debited from channel but not received at destination wallet

### ISSUE THREAD

**Issue by russellsingapore** [ACINQ/eclair#1000]
I am using Eclair version v0.2-beta9 and have it set up as a full lightning node. Everything has been working fine. However I sent a 250,000 Satoshi transaction as a test to my Blue Wallet and the funds have not arrived. I can see that the funds have been taken from one of the Channels and there is no record of the transaction on the Activity tab. I believe that this should be a technical issue that needs addressing as loss of funds is quite serious. Details are below.

The invoice for the transaction is: lnbc2500u1pwd0kvvpp5nr2fnr3fzlxhgzkyd093jh9e27uek6w8cqmn9t06znxx9vwajcfsdqqcqzpglxhh0gm77dfnk808sknjxtr96mvu7fnf0vlars8f55qvu9t59wcqsjvqqu6ehjtgqjk0zhgquw7fcy3rt23jpgun9rucy9evq5cvrdqps40t23

My Lightning node details are:
03afdcba5bf546ddde69a28d55476b0be0d3d2a673979c6b37b1e0366a768485b3@86.3.45.243:9735

Transaction took place at approx 9:35 am on 12/5/2019 (UK time)

Further information in attached file

**Comment by russellsingapore** [comment]
<img width="945" alt="Capture" src="https://user-images.githubusercontent.com/22152144/57581398-db118200-74ae-11e9-936c-e64278dc1c66.PNG">

**Comment by pm47** [comment]
If eclair told you that the payment has been sent, then it proves that BlueWallet has received the payment and the issue is on their side.

Can you provide your logs?

**Comment by russellsingapore** [comment]
Eclair has not told me that payment has been sent. There is nothing in the activity tab. Also when I go through the log file it is very big and all I can see is that there were many failed attempts. What do I need to look for in the log file as there were many attempts to push this through before it appeared to actually go through. I have checked my Blue Wallet and it shows that the actual invoice has expired and payment was not received. However my total Bitcoin balance on Eclair has gone down by the 250,000 Satoshi amount and the channel in question is showing at 50% with my side of the balance down by 250,000.  This should not be a blame game in that the problem lies with someone else as the lightning network has clearly failed here and we need to know why.

**Comment by pm47** [comment]
> What do I need to look for in the log file as there were many attempts to push this through before it appeared to actually go through.

Filter for lines containing `channel.Channel`, there shouldn't be too many of them. Alternatively, [enable the API](https://github.com/ACINQ/eclair/wiki/API), and paste the result of the [`channels`](https://acinq.github.io/eclair/#channels) call.

> This should not be a blame game in that the problem lies with someone else as the lightning network has clearly failed here and we need to know why.

Not playing that game, just trying to save my time. What probably happened is that your payment is pending because of some faulty intermediate node in the route which is stuck. The payment will eventually timeout, and you will get your funds back. If the faulty node is your direct peer this will lead to the channel being closed as part of the refund process. There are a lot of unreliable nodes out there, and still very annoying bugs.

**Comment by russellsingapore** [comment]
Thanks - I have searched on channel.Channel and I can see lots of these around the time of the transaction. However I don't know what I am really trying to see. I can see there are sections which look different. Can I send the log file for you to see?

**Comment by russellsingapore** [comment]
Here is the log file - the transaction was sometime after 9.30am - 
[eclair.log](https://github.com/ACINQ/eclair/files/3170411/eclair.log)

**Comment by pm47** [comment]
Ok, you indeed have one payment pending for 250001750 msats:

`
2019-05-12 09:33:45,766 INFO  f.a.eclair.Diagnostics  n:0330f0afe893e809d8790b26b40d9353f586ef0894d4ae704d06be24538c030457 c:9d4053be477970b8da129cd31bf8bbb9a2be613387d5caf4f040bab4a366b032 - OUT msg=UpdateAddHtlc(9d4053be477970b8da129cd31bf8bbb9a2be613387d5caf4f040bab4a366b032,7,250001750,98d4998e2917cd740ac46bcb195cb957b99b69c7c03732adfa14cc62b1dd9613,576266,<onion>)
`

Route was:
- 0330f0afe893e809d8790b26b40d9353f586ef0894d4ae704d06be24538c030457
- 03bb88ccc444534da7b5b64b4f7b15e1eccb18e102db0e400d4b9cfe93763aa26d
- 031d244a60d61936834f290656cdf0ed4b9979260c19b07804bde681819f8e14ca
- 03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f
- 03021c5f5f57322740e4ee6936452add19dc7ea7ccf90635f95119ab82a62ae268

As it happens, the second to last node is ours (`03864e...`), I checked the log and your payment didn't reach it, so the faulty node is either `03bb88` or `031d24`.

Bottom line: unfortunately there isn't much to do except waiting for your payment to expire. This will happen at block `576266`, a little less than 4 days from now.

**Comment by russellsingapore** [comment]
Thanks for looking into that - does that mean that the payment will come back to me in 4 days and go back into the original channel?

**Comment by pm47** [comment]
Depending on which peer is faulty, it will either go back to the original channel, or that channel will get closed and you will have two incoming on-chain transactions (one for your main balance, one for the payment itself).

**Comment by russellsingapore** [comment]
Ok - thanks - I will update this post once I see what happens

**Comment by russellsingapore** [comment]
Just to let you know that the payment went back into my Eclair wallet. I got an error pop up saying payment failed and the channel that was missing the funds was re credited - thanks for the help.

**Comment by pm47** [comment]
Great. So it's definitely not your direct peer which was faulty. This also explains why it got resolved earlier than 4 days: timeouts increase for everyhop from the destination to the origin.
